### PR TITLE
Add support for FMCOMMS11, AD9625, AD9162

### PR DESCRIPTION
**Pull request by HRosete**
# Description

Adding support to the following devices:
- **AD9625** 
1. Appropriated the _rx_channel_names with its iio_info (deleted voltage1 because it's not in iio_info and this causes problems)
2. Added rx_sample_rate property
3. Add FMCOMMS11 to its test's hardware

- **AD9162**
1. Set _complex_data to False to cater FMCOMMS11
2. Rename sampling_rate to sample_rate to cater to dma_tests
3. Deleted unnecessary comments
4. Add FMCOMMS11 to its test's hardware

- **FMCOMMS11**
1. Class abstraction
2. Documentation
3. Example
4. Test script 
5. Added projects to test_map.py
6. (already has an emulated file in pytest-libiiio)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How has this been tested?

Test Pipelines:
[HW_pyadi_test/229](http://10.116.110.50/jenkins/blue/organizations/jenkins/HW_tests%2FHW_pyadi_test/detail/HW_pyadi_test/229/pipeline)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

@tfcollins

### CODE REVIEW

**Comment by HRosete** [comment]
An issue came up as I was testing my code: AD9162's (FMCOMMS11's DAC) TX channels can't be enabled; tried to rename them to the usual TX channel names convention but still, they can't be found. Should I take out FMCOMMS11's pytest from the commits in the meantime? Any help on this issue is really appreciated.

**Review by tfcollins — CHANGES_REQUESTED** [review]

File: adi/adl5240.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from adi.context_manager import context_manager
+from adi.rx_tx import tx
+from adi.sync_start import sync_start
+
+
+class adl5240(tx, context_manager, sync_start):
```
[the remark this answers is not in the document]
Comment (HRosete) [reply adi/adl5240.py:39]: adl5240-iio_info
adl5240-channel-and-attribute

It has a channel (voltage0) and has 1 channel-specific attribute. Other than that, there's none. I believe it doesn't have any TX DMA buffers on it. Okay, I can now see why. I'll be editing the code.

[the remark this answers is not in the document]
Comment (HRosete) [reply adi/adl5240.py]: Please review commit [bfb81bd](https://github.com/analogdevicesinc/pyadi-iio/pull/333/commits/bfb81bd0f286ddc1ccdbfec78833b12fd576d02d).

File: adi/adl5240.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    _complex_data = False
+    _tx_channel_names = ["voltage0"]
+    _device_name = ""
+
+    def __init__(self, uri=""):
+
+        context_manager.__init__(self, uri, self._device_name)
+
+        self._txdac = self._ctx.find_device("adl5240")
```
Comment (tfcollins) [inline adi/adl5240.py:49]: Switch to 
```
self._ctrl = self._ctx.find_device("adl5240")
```
Otherwise the attribute read/writes won't work

Comment (HRosete) [reply adi/adl5240.py]: Changed with commit [1d98ca4](https://github.com/analogdevicesinc/pyadi-iio/pull/333/commits/1d98ca4b2a2a41009e7c120648d6714f09589218).

**Comment by HRosete** [comment]
I will have to merge my edits with AD9162 with the current support for the AD916x family in the repository. I'll have to revert this back to draft.

**Comment by HRosete** [comment]
I will be dropping the support for AD9172 in this PR as well as the renaming of the existing AD916x module to AD9166 to their own corresponding PRs: PR #365 for AD917 and PR #366 for renaming AD916x to AD9166.

**Comment by HRosete** [comment]
> I will be dropping the support for AD9172 in this PR as well as the renaming of the existing AD916x module to AD9166 to their own corresponding PRs: PR #365 for AD917 and PR #366 for renaming AD916x to AD9166.

Support for AD9172 - [9546ece](https://github.com/analogdevicesinc/pyadi-iio/pull/333/commits/9546ece1be2ec6b9a50417c53e83a9ca58a0b841)
Renaming AD916x - [985febc](https://github.com/analogdevicesinc/pyadi-iio/pull/333/commits/985febc547ace68e05f2a9c7ec69fe79693694fd)

**Comment by HRosete** [comment]
Dropping commits for support for these devices:
- [Add support for ADF4355](https://github.com/analogdevicesinc/pyadi-iio/pull/333/commits/4483a96232e538adb1198d09529db4aa9c2784ed)
- [Add support for ADL5240](https://github.com/analogdevicesinc/pyadi-iio/pull/333/commits/2009688e120baac6ff2269f199bedac5cf5663de)
- [Add support for AD9162](https://github.com/analogdevicesinc/pyadi-iio/pull/333/commits/597fee8d6e7d0fb0a13d3fb5c015bdece9143c4e)

These classes are working apart from AD9625. Putting these commits and supports in PR #379

**Comment by HRosete** [comment]
Ran the pre-commit and it showed no errors. I don't know where to look for the errors in the Documentation.

**Comment by tfcollins** [comment]
Please rebase on master. Older 3.6 build were removed

**Comment by HRosete** [comment]
This PR is now rebased to master.

File: adi/fmcomms11.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from adi.ad9162 import ad9162
+from adi.ad9625 import ad9625
+
+# from adi.adl5240 import adl5240
```
Comment (tfcollins) [inline adi/fmcomms11.py:37]: Can you remove the commented adl5240 references

**Review by tfcollins — CHANGES_REQUESTED** [review]

File: adi/fmcomms11.py (hunk 3 above)
Comment (HRosete) [reply adi/fmcomms11.py]: Hello Travis, I removed them in commit 97e9c3d.

**Comment by HRosete** [comment]
Squashed the changes in commit [0ea6d06](https://github.com/analogdevicesinc/pyadi-iio/pull/333/commits/0ea6d06a8c2675648626e62f973a6922aae407f3) to clean the commit history.

**Pull request merged**